config_gen.py

```diff
@@ -303,6 +289,42 @@ def run(cmd, *args, **kwargs):
         print("\nRunning make...")
         run(make_args, env=env, **proc_opts)
 
+    elif(os.path.exists(os.path.join(project_dir, "Make/options"))):
+        # Added for supporting wmake (OpenFOAM build system) 2016/03/18
+        print("Found OpenFOAM Make/options")
+        print("Preparing build directory...")
+        build_dir = tempfile.mkdtemp()
+
+        # wmake uses environment variables starting with WM_
+        # since OpenFOAM 3.x.x, there were some variables added.
+        keys = os.environ.keys()
+        WM = re.compile('^WM_')
+        for key in keys:
+            if(WM.match(key)):
+                env[key] = os.environ[key]
```

Comment:
As mentioned above, this section should be removed / replaced with an error if the environment variables are not already present in `env`.